src/api/materials.py

```
@@ -0,0 +1,541 @@
+    if not file_field:
+        return w.err("file is required", 400)
+
+    # Support both dict (multipart/JSON) and raw bytes (test injection)
+    if isinstance(file_field, dict):
+        filename = file_field.get("filename") or "upload"
+        raw_bytes = file_field.get("bytes") or b""
+        # JSON fallback may provide a list of ints; normalise to bytes.
+        if isinstance(raw_bytes, list):
+            try:
+                file_bytes = bytes(raw_bytes)
+            except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)